usecases/traverser/aggregate_type_inspector.go

```diff
@@ -15,18 +15,18 @@ import (
 	"fmt"
 
 	"github.com/weaviate/weaviate/entities/aggregation"
+	"github.com/weaviate/weaviate/entities/models"
 	"github.com/weaviate/weaviate/entities/schema"
 	schemaUC "github.com/weaviate/weaviate/usecases/schema"
 )
 
 type typeInspector struct {
-	schema schema.Schema
+	getClass func(string) *models.Class
 }
 
 func newTypeInspector(schemaGetter schemaUC.SchemaGetter) *typeInspector {
-	schema := schemaGetter.GetSchemaSkipAuth()
 	return &typeInspector{
-		schema: schema,
+		getClass: schemaGetter.ReadOnlyClass,
```

Comment:
**suggestion:**  i think we don't need the whole `schemaUC.SchemaGetter` interface we could directly pass the get func to avoid passing nil